include/kernels/gauge_loop_trace.cuh

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include <gauge_field_order.h>
+#include <quda_matrix.h>
+#include <index_helper.cuh>
+#include <kernel.h>
+#include <thread_array.h>
+#include <array.h>
+#include <reduce_helper.h>
+#include <reduction_kernel.h>
+#include <gauge_path_helper.cuh>
+
+namespace quda {
+
+  /**
+    @brief Return the batch block size used for multi reductions.
+  */
+  constexpr unsigned int max_n_batch_block_loop_trace() { return 8; }
+
+  template <typename Float_, int nColor_, QudaReconstructType recon_>
+  struct GaugeLoopTraceArg : public ReduceArg<array<double, 2>>  {
+    using Float = Float_;
```

Comment:
As mentioned elsewhere: change this to using `real = typename mapper<Float>::type;` and change the template type to `store_t`

Reply:
https://github.com/lattice/quda/commit/58845f78894fa8a26a831a6d0a61b8a50b56a269